BD-3702 Add Contentful partner docs (#8530)
braze-inc/braze-docs: _docs/_partners/message_personalization/dynamic_content/contentful.md

---
nav_title: Contentful
article_title: Contentful

> description: "How to use Connected Content to dynamicallyvpull content from Contentful into your Braze campaigns."

Reviewer: ```suggestion
description: "This reference article outlines the partnership between Braze and Contentful, a content management system that allows you to dynamically use Connected Content to pull content from Contentful into your Braze campaigns."
```

alias: /partners/contentful/
page_type: partner
search_tag: Partner

> layout: dev_guide

Reviewer: ```suggestion
```
Removing this YAML bit so the "Edit this page on GitHub" header nav appears

---

# Contentful

> >[Contentful](https://www.contentful.com/) is a headless content management system that lets you create, manage, and distribute content to any platform. Unlike a content management system (CMS), Contentful give you total freedom to create your own content model so you can decide which content you want to manage.<br><br>This page provides a step-by-step guide to configure Braze Connected Content to fetch data from Contentful's Content Delivery API. 

Reviewer: ```suggestion
>[Contentful](https://www.contentful.com/) is a headless content management system that lets you create, manage, and distribute content to any platform. Unlike a content management system (CMS), Contentful allows you to create your content model so you can decide which content you want to manage.<br><br>This page provides a step-by-step guide to configure Braze Connected Content to fetch data from Contentful's Content Delivery API. 
```

After you're integrated, you can use Contentful's RESTful APIs to deliver your content across multiple channels, such as websites, mobile apps (iOS, Android, and Windows), or many other platforms. You can also dynamically pull content from Contentful for use in your Braze campaigns.

## Prerequisites

Before you start, you'll need the following:

| Prerequisite          | Description                        |
|-----------------------|------------------------------------|
| A Contentful account | You need a Contentful account with access to the Content Delivery API. |

> | A Braze Account | You need a Braze account with access to the Connected Content feature. |

Reviewer: ```suggestion
| A Braze account | You need a Braze account with access to the Connected Content feature. |
```

{: .reset-td-br-1 .reset-td-br-2 role="presentation" }

{% alert note %}

> If you are using the [older navigation]({{site.baseurl}}/navigation/), you can create an API key at **Developer Console** > **API Settings**.

Reviewer: ```suggestion
If you're using the [older navigation]({{site.baseurl}}/navigation/), you can create an API key at **Developer Console** > **API Settings**.
```

{% endalert %}

## Integration

> ### Step 1: Obtain Contentful API credentials

Reviewer: ```suggestion
### Step 1: Get your Contentful API credentials
```
Minor suggestion, feel free to disregard :)

1. [Log into Contentful](https://app.contentful.com/login) with your credentials.

> 2. Create or retrieve API Access Tokens in the Contentful dashboard by going to **Settings** > **API keys**. If you don't already have an API key, create a new one:<br>2.1 Select **Add API key**.<br>2.2 Enter the required details and select the appropriate environment.<br>2.3 Select **Save** and note the **Space ID** and **Content Delivery API - access token**.

Reviewer: ```suggestion
2. Create or retrieve API access tokens in the Contentful dashboard by going to **Settings** > **API keys**. If you don't already have an API key, create a new one:<br>2.1 Select **Add API key**.<br>2.2 Enter the required details and select the appropriate environment.<br>2.3 Select **Save** and note the **Space ID** and **Content Delivery API - access token**.
```

{% alert note %}
Different endpoints may require new variables, for instance the Images API requires an `{asset_id}`, `{unique_id},` and `{name}`. For further guidance, contact Contentful.
{% endalert %}

{% raw %}
```json
        {% assign space_id = "YOUR-CONTENTFUL-SPACE-ID"}
        {% assign environment_id = "YOUR-CONTENTFUL-ENVIRONMENT-ID"}
        {% assign entry_id = "YOUR-CONTENTFUL-ENTRY-ID"}
        {% assign access_token = "YOUR-CONTENTFUL-ACCESS-TOKEN"}
         {% connected_content https://cdn.contentful.com/spaces/{space_id}/environments/{environment_id}/entries/{entry_id}?access_token={access_token}
         :method get
         :headers {
             "Authorization": "YOUR_CONTENTFUL_ACCESS_TOKEN"
                 }
               :content_type application/json
               :save response %}
```
{% endraw %}

{: start="4"}
4. Use "Test Endpoint" to test that Braze can successfully connect to the Contentful API and retrieve the desired data.

> 5. After testing the endpoint and confirming that it works, select **Done** to save your Content Block.

Reviewer: ```suggestion
5. Sselect **Done** to save your Content Block.
```

> 6. Give your Content Block a descriptive name, such as "Contentful API". Then select **Launch Content Block**.

Reviewer: ```suggestion
6. Give your Content Block a descriptive name, such as "Contentful API", then select **Launch Content Block**.
```

{% raw %}
```json
   {
  "fields": {
    "title": {
      "en-US": "Hello!"
    },
    "body": {
      "en-US": "This is a sample message!"
    }
  },
  "metadata": {
    "tags": [
      {
        "sys": {
          "type": "Link",
          "linkType": "Tag",
          "id": "nyCampaign"
        }
      }
    ]
  },
  "sys": {
    "id": "5KsDBWseXY6QegucYAoacS",
    "type": "Entry",
    "version": 1,
    "space": {
      "sys": {
        "type": "Link",
        "linkType": "Space",
        "id": "yadj1kx9rmg0"
      }
    },
    "contentType": {
      "sys": {
        "type": "Link",
        "linkType": "ContentType",
        "id": "hfM9RCJIk0wIm06WkEOQY"
      }
    },
    "createdAt": "2016-12-20T10:43:35.772Z",
    "updatedAt": "2016-12-20T10:43:35.772Z",
    "revision": 1
  }
}
```
{% endraw %}

{: start="3" }

> 3. Preview and test your campaign to confirm that the Connected Content data correctly displays.
> 4. After you're satisfied with the setup, launch your campaign.

Reviewer: ```suggestion
3. Preview and test your campaign to confirm that the Connected Content data displays correctly.
4. After you're satisfied with the setup, launch your campaign.
```

## Troubleshooting

> ### API response issues
> 
> Make sure that your Contentful API credentials and endpoint URL are correct. Check for any error messages in Braze that might indicate issues with the API call.
> 
> ### Data mapping problems

Reviewer: ```suggestion
### API response

Make sure that your Contentful API credentials and endpoint URL are correct. Check for any error messages in Braze that might indicate issues with the API call.

### Data mapping
```
minor suggestion, feel free to disregard
